jabgui/src/main/java/org/jabref/gui/mergeentries/UpdateEntryFromWebSourcesAction.java

```diff
@@ -0,0 +1,125 @@
+        if (stateManager.getActiveDatabase().isEmpty()) {
+            return;
+        }
+
+        if (stateManager.getSelectedEntries().size() != 1) {
+            notificationService.notify(Localization.lang("Select exactly one entry to update from web sources"));
+            return;
+        }
+        BibEntry entry = stateManager.getSelectedEntries().getFirst();
+
+        MultiMergeEntriesView view = new MultiMergeEntriesView(preferences, taskExecutor);
+        view.addSource(Localization.lang("Entry"), entry);
+
+        SortedSet<EntryBasedFetcher> fetchers = WebFetchers.getEntryBasedFetchers(preferences.getImporterPreferences(), preferences.getImportFormatPreferences(), preferences.getFilePreferences(), stateManager.getActiveDatabase().get());
+
+        for (EntryBasedFetcher fetcher : fetchers) {
+            view.addSource(fetcher.getName(), () -> {
+                try {
+                    List<BibEntry> results = fetcher.performSearch(entry);
+                    if (results.isEmpty()) {
+                        return null;
+                    }
+                    return results.getFirst();
+                } catch (FetcherException e) {
+                    return null;
+                }
+            });
+        }
+        Optional<BibEntry> merged = dialogService.showCustomDialogAndWait(view);
+
+        if (merged.isEmpty()) {
+            notificationService.notify(Localization.lang("Canceled updating entry from web sources"));
+            return;
+        }
+
+        BibEntry mergedEntry = merged.get();
+        NamedCompoundEdit compoundEdit = new NamedCompoundEdit(Localization.lang("Update entry from web sources"));
+
+        Set<Field> jointFields = new TreeSet<>(Comparator.comparing(Field::getName));
+        jointFields.addAll(mergedEntry.getFields());
+        Set<Field> originalFields = new TreeSet<>(Comparator.comparing(Field::getName));
+        originalFields.addAll(entry.getFields());
+
+        boolean edited = false;
+
+        for (Field field : jointFields) {
+            Optional<String> originalString = entry.getField(field);
+            Optional<String> mergedString = mergedEntry.getField(field);
+
+            if (originalString.isEmpty() || !originalString.equals(mergedString)) {
+                String newValue = mergedString.orElse("");
+                entry.setField(field, newValue);
+                compoundEdit.addEdit(new UndoableFieldChange(entry, field, originalString.orElse(null), newValue));
+                edited = true;
+            }
+        }
+
+        for (Field field : originalFields) {
+            if (!jointFields.contains(field) && !FieldFactory.isInternalField(field)) {
+                Optional<String> originalString = entry.getField(field);
+                entry.clearField(field);
+                compoundEdit.addEdit(new UndoableFieldChange(entry, field, originalString.orElse(null), null));
+                edited = true;
+            }
+        }
```

Comment:
This is AI genereted, isn't it? Our merge dialogs should already be able to do it.